src/mrpro/operators/_CartesianSamplingOp.py

```diff
@@ -78,9 +80,37 @@ def __init__(self, encoding_matrix: SpatialDimension[int], traj: KTrajectory) ->
         # 1D indices into a flattened tensor.
         kidx = kz_idx * sorted_grid_shape.y * sorted_grid_shape.x + ky_idx * sorted_grid_shape.x + kx_idx
         kidx = rearrange(kidx, '... kz ky kx -> ... 1 (kz ky kx)')
+
+        # check that all points are inside the encoding matrix
+        self._data_outside_of_encoding_matrix = False
+        inside_encoding_matrix = (
+            (kx_idx >= 0)
+            & (kx_idx < sorted_grid_shape.x)
+            & (ky_idx >= 0)
+            & (ky_idx < sorted_grid_shape.y)
+            & (kz_idx >= 0)
+            & (kz_idx < sorted_grid_shape.z)
+        )
+        if torch.any(~inside_encoding_matrix):
+            warnings.warn(
+                'K-space points lie outside of the encoding_matrix and will be ignored.'
+                ' Increase the encoding_matrix to include these points.',
+                stacklevel=2,
+            )
+
+            inside_encoding_matrix = rearrange(inside_encoding_matrix, '... kz ky kx -> ... 1 (kz ky kx)')
```

Comment:
This part is not very nice. Maybe somebody has a better idea of how to do create the indices.

Reply:
i have a suggestion using non-zero